### PR TITLE
Fix warnings about casting pointer from/to different size integer

### PR DESCRIPTION
**Pull request by zhekov**
Note that the PR is tested under Win64 only. If warnings appear under Win32 or Unix, we will need to find even more compatible integer types.

### CODE REVIEW

**Comment by zhekov** [comment]
My options are "-O2 -Wall -Wextra -Wno-deprecated-declarations -Wno-unused-parameter", and these were all warnings containing "integer". Is there an extra option to warn about precision loss on integer conversions?

**Comment by elextr** [comment]
A few I could find that *might* catch messy things which I don't think are in Wall or Wextra: -Wno-int-conversion -Wno-int-to-pointer-cast and -Wno-pointer-to-int-cast -Wconversion.

File: src/win32.c [hunk 1]
```diff
@@ -816,7 +816,7 @@ static FILE *open_std_handle(DWORD handle, const char *mode)
 	if (hConHandle == -1)
 	{
 		gchar *err = g_win32_error_message(GetLastError());
-		g_warning("_open_osfhandle(%ld, _O_TEXT) failed: %s", (long)lStdHandle, err);
+		g_warning("_open_osfhandle(handle(%ld), _O_TEXT) failed: %s", (long)handle, err);
```
Comment (b4n) [inline src/win32.c]: why this change?

**Comment by b4n** [comment]
Otherwise LGTM

File: src/win32.c (hunk 1 above)
Comment (zhekov) [inline src/win32.c]: Because (a) the lStdHandle value does not carry any useful information, it may be anything, while handle is at least the constant numeric value of STD_INPUT/OUTPUT/ERROR_HANDLE, and (b) if a lStdHandle is used, it should at least be cast properly, with HandleToLong() or something, instead of (long). Only the least significant 32-bits are actually used, so "%ld" is OK.

Comment (b4n) [inline src/win32.c]: OK
